RabbitMQ/PhpAmqp Driver

rtuin opened the pull request:
Hi there,

here is the driver that I've built to use Bernard with RabbitMQ/PhpAmqp.

The php-amqp lib receives messages from a consumer via a callback (async), which clashes a bit with  Bernard's `popMessage` (sync). The workaround in this driver turns out to be hard to test. Suggestions are more than welcome!

Furthermore it is important that the QoS on the channel is set to have a maximum of one unacked message at a time.

Shouts out to @rskuipers for helping me with this.

Possibly solves #115 

## Todo
- [x] Add documentation
- [x] Fix reason CI build failed

dennisdegreef commented on src/Driver/PhpAmqpDriver.php:
```diff
@@ -0,0 +1,181 @@
+
+    /**
+     * @param AMQPConnection $connection
+     * @param $exchange string
```
Wrong order of parameters :)

rtuin replied:
Good catch, thanks!

henrikbjorn commented:
Can you rebase this?

rtuin commented:
@henrikbjorn I gained some new insights on how to make this driver more efficient. I'll incorporate them into the code this week and rebase in the process.

rtuin commented:
@henrikbjorn I finally found some times to port the optimisations to this PR. This also drastically improves the driver's test coverage!
Details on this are in the last commit.

henrikbjorn commented:
Look really cool.

Only thing is this introduces a minor issue accordance to Scrutinizer https://scrutinizer-ci.com/g/bernardphp/bernard/inspections/4f01d8ca-3744-4f45-83bf-714d5665175e/issues/files/src/Driver/PhpAmqpDriver.php?status=new&orderField=path&order=asc

rtuin commented:
Nice catch! I'm impressed scrutinizer found that.
I've updated the code and rebased, but the new commit doesn't appear here for me. Sure hope it's a GitHub cache issue.

rtuin commented:
Squashed the commits into one now. The scrutinizer message changes, but an issue remains. At least the scrutinizer points to the PhpAmqpLib now.

henrikbjorn commented:
Thanks :)
